File: employee_project/src/main/java/com/ha/java/services/EmployeeService.java
package com.ha.java.services;

import com.ha.java.dto.Employee;
import com.ha.java.dto.EmployeeDetailsPageable;
import com.ha.java.entity.EmployeeEntity;

public interface EmployeeService {
    EmployeeEntity createEmployee(EmployeeEntity employee);


    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployee(Long id, Employee employee);

    //search
    EmployeeDetailsPageable findByAny(String anyField, String sortBy, String direction, int page, int size);

}

File: employee_front/src/service/EmployeeService.js
import axios from "axios"

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/employees"

const headers = {
    headers:{
        'Authorization': "Bearer " + localStorage.getItem("token")
    }
}


class EmployeeService{
    
    saveEmployee(employee){
        return axios.post(EMPLOYEE_API_BASE_URL, employee, headers);
    }

    getEmployees(search, sortBy, page, direction){
        return axios.get(`http://localhost:8080/employees/search?search=${search}&sortBy=${sortBy}&page=${page}&size=5&direction=${direction}` ,
        headers);
    }

    deleteEmployee(id){
        return axios.delete(EMPLOYEE_API_BASE_URL + "/" + id, headers)
    }

    // deleteEmployee(id, headers) {
    //     return axios.delete(EMPLOYEE_API_BASE_URL + "/" + id, { headers });
    // }
    

    getEmployeeById(id){
        return axios.get(EMPLOYEE_API_BASE_URL + "/" + id, headers)
    }

    updateEmployee(id, employee){
        return axios.put(EMPLOYEE_API_BASE_URL + "/" + id, employee, headers)
    }

    searchEmployee(name, page){
        return axios.get(`http://localhost:8080/employees/search?search=${name}&page=${page}&size=5`, headers)
    }


}
export default new EmployeeService();